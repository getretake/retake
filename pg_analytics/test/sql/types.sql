-- Supported Types
CREATE TABLE t (a text) USING parquet;
INSERT INTO t VALUES ('hello world');
SELECT * FROM t;
DROP TABLE t;

CREATE TABLE t (a varchar) USING parquet;
INSERT INTO t VALUES ('hello world');
SELECT * FROM t;
DROP TABLE t;

CREATE TABLE t (a char) USING parquet;
INSERT INTO t VALUES ('h');
SELECT * FROM t;
DROP TABLE t;

CREATE TABLE t (a smallint) USING parquet;
INSERT INTO t VALUES (1);
SELECT * FROM t;
DROP TABLE t;

CREATE TABLE t (a integer) USING parquet;
INSERT INTO t VALUES (1);
SELECT * FROM t;
DROP TABLE t;

CREATE TABLE t (a bigint) USING parquet;
INSERT INTO t VALUES (1);
SELECT * FROM t;
DROP TABLE t;

CREATE TABLE t (a real) USING parquet;
INSERT INTO t VALUES (1.0);
SELECT * FROM t;
DROP TABLE t;

CREATE TABLE t (a double precision) USING parquet;
INSERT INTO t VALUES (1.0);
SELECT * FROM t;
DROP TABLE t;

CREATE TABLE t (a bool) USING parquet;
INSERT INTO t VALUES (true);
SELECT * FROM t;
DROP TABLE t;

CREATE TABLE t (a numeric(5, 2)) USING parquet;
INSERT INTO t VALUES (1.01);
SELECT * FROM t;
DROP TABLE t;

CREATE TABLE t (a timestamp) USING parquet;
INSERT INTO t VALUES ('2024-01-29 15:30:00');
SELECT * FROM t;
DROP TABLE t;

CREATE TABLE t (a date) USING parquet;
INSERT INTO t VALUES ('2024-01-29');
SELECT * FROM t;
DROP TABLE t;

-- Unsupported Types
CREATE TABLE t (a bytea) USING parquet;
CREATE TABLE t (a uuid) USING parquet;
CREATE TABLE t (a oid) USING parquet;
CREATE TABLE t (a json) USING parquet;
CREATE TABLE t (a jsonb) USING parquet;
CREATE TABLE t (a time) USING parquet;
CREATE TABLE t (a timetz) USING parquet;
CREATE TABLE t (a text[]) USING parquet;
