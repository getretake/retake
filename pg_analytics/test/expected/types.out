-- Supported Types
CREATE TABLE t (a text) USING deltalake;
INSERT INTO t VALUES ('hello world');
SELECT * FROM t;
      a      
-------------
 hello world
(1 row)

DROP TABLE t;
CREATE TABLE t (a varchar) USING deltalake;
INSERT INTO t VALUES ('hello world');
SELECT * FROM t;
      a      
-------------
 hello world
(1 row)

DROP TABLE t;
CREATE TABLE t (a char) USING deltalake;
INSERT INTO t VALUES ('h');
SELECT * FROM t;
 a 
---
 h
(1 row)

DROP TABLE t;
CREATE TABLE t (a smallint) USING deltalake;
INSERT INTO t VALUES (1);
SELECT * FROM t;
 a
---
 1
(1 row)

DROP TABLE t;
CREATE TABLE t (a integer) USING deltalake;
INSERT INTO t VALUES (1);
SELECT * FROM t;
 a
---
 1
(1 row)

DROP TABLE t;
CREATE TABLE t (a bigint) USING deltalake;
INSERT INTO t VALUES (1);
SELECT * FROM t;
 a
---
 1
(1 row)

DROP TABLE t;
CREATE TABLE t (a real) USING deltalake;
INSERT INTO t VALUES (1.0);
SELECT * FROM t;
 a
---
 1
(1 row)

DROP TABLE t;
CREATE TABLE t (a double precision) USING deltalake;
INSERT INTO t VALUES (1.0);
SELECT * FROM t;
 a 
---
 1
(1 row)

DROP TABLE t;
CREATE TABLE t (a bool) USING deltalake;
INSERT INTO t VALUES (true);
SELECT * FROM t;
 a 
---
 t
(1 row)

DROP TABLE t;
CREATE TABLE t (a numeric(5, 2)) USING deltalake;
INSERT INTO t VALUES (1.01);
SELECT * FROM t;
  a   
------
 1.01
(1 row)

DROP TABLE t;
CREATE TABLE t (a timestamp) USING deltalake;
INSERT INTO t VALUES ('2024-01-29 15:30:00');
SELECT * FROM t;
          a          
---------------------
 2024-01-29 15:30:00
(1 row)

DROP TABLE t;
CREATE TABLE t (a date) USING deltalake;
INSERT INTO t VALUES ('2024-01-29');
SELECT * FROM t;
     a      
------------
 2024-01-29
(1 row)

DROP TABLE t;
-- Unsupported Types
CREATE TABLE t (a bytea) USING deltalake;
ERROR:  Postgres type BYTEAOID not supported

CREATE TABLE t (a uuid) USING deltalake;
ERROR:  Postgres type UUIDOID not supported

CREATE TABLE t (a oid) USING deltalake;
ERROR:  Postgres type OIDOID not supported

CREATE TABLE t (a json) USING deltalake;
ERROR:  Postgres type JSONOID not supported

CREATE TABLE t (a jsonb) USING deltalake;
ERROR:  Postgres type JSONBOID not supported

CREATE TABLE t (a time) USING deltalake;
ERROR:  Postgres type TIMEOID not supported

CREATE TABLE t (a timetz) USING deltalake;
ERROR:  Postgres type TIMETZOID not supported

CREATE TABLE t (a text[]) USING deltalake;
ERROR:  Arrays are not yet supported
